lib/reek/context/module_context.rb

```diff
@@ -75,9 +75,13 @@ def descriptively_commented?
       #
       # @return true if the module is a namespace module
       def namespace_module?
-        return false if exp.type == :casgn
-        contents = exp.children.last
-        contents && contents.find_nodes([:def, :defs], [:casgn, :class, :module]).empty?
+        children = direct_children
+        children.any? && children.all? { |child| [:casgn, :class, :module].include? child.type }
+      end
+
+      def direct_children
+        contents = exp.children.last or return []
```

Comment:
How about having a comment that gives a short, concise example of the different AST's we have to deal with here?